openlibrary/plugins/openlibrary/js/initAnnouncementBanner.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Init the announcement banners if any
+ * If patron has dismissed the banner by clicking the 'X' button, banner storage key will be added to localstorage
+ * Banners with storage keys within the localstorage will not show again
+ */
+function addBannerStorageKeyToUserPreferences(bannerKey, success) {
+    $.ajax({
+        type: 'POST',
+        url: '/hide_banner',
+        data: JSON.stringify({'storage-key': bannerKey}),
+        contentType: 'application/json',
+        dataType: 'json',
+
+        beforeSend: function(xhr) {
+            xhr.setRequestHeader('Content-Type', 'application/json');
+            xhr.setRequestHeader('Accept', 'application/json');
+        },
+        success: success
+    });
+}
+
+export function initAnnouncementBanner(announcementBanner) {
+    const bannerKey = announcementBanner.getAttribute('data-storage-key');
```

Comment:
```suggestion
    const bannerKey = announcementBanner.dataset.storageKey
```
We use the `dataset` property to get data attributes elsewhere in the codebase.  [Here](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes#javascript_access) are some more details about `dataset`.